vignettes/getting-started-part-I.Rmd

```diff
@@ -0,0 +1,220 @@
+
+```{r}
+model <- enw_model(threads = TRUE)
+```
+
+# Bringing it all together: Fitting the model
+
+We can now fit the model using the ["No-U-Turn Sampler Markov chain Monte Carlo" method](https://mc-stan.org/docs/reference-manual/hamiltonian-monte-carlo.html). This is a type of Hamiltonian Monte Carlo (HMC) algorithm and is the core fitting method used by `cmdstan`. The NUTS MCMC method is efficient, automatically tunes its own parameters and is robust to correlations between parameters, making it fast and effective at generating samples from the posterior distribution. We specify fitting options using `enw_fit_opts()` (note that the settings shown here are tuned for speed and may not be appropriate for many real world use cases). We also pass our preprocessed data (`pobs`), our pre-compiled model (`model`), and our model modules (`expectation_module`, `reference_module`, and `report_module`) to `epinowcast`, where they are combined and used to fit the model.
+
+```{r}
+options(mc.cores = 2)
+nowcast <- epinowcast(data = pobs,
+  expectation = expectation_module,
+  reference = reference_module,
+  report = report_module,
+  fit = enw_fit_opts(
+    save_warmup = FALSE, pp = TRUE,
+    chains = 2, threads_per_chain = 2,
+    iter_sampling = 500, iter_warmup = 500,
+    show_messages = FALSE
+  ),
+  model = model
+)
+```
+
+# The `epinowcast` object
+
+The `epinowcast()` function returns an `epinowcast` object which includes diagnostic information, the data used for fitting, and the underlying [`CmdStanModel` object](https://mc-stan.org/cmdstanr/reference/CmdStanModel.html).
+
+```{r}
+nowcast
+```
+
+# Summarising and plotting the nowcast
+
+The nowcast (the combination of currently observed and predicted unobserved data) can then be summarised using
+
+```{r}
+nowcast |>
+  summary(probs = c(0.05, 0.95)) |>
+  head(n = 10)
+```
+
+Similarly, the summarised nowcast can be plotted against the latest observed data using
+
+```{r nowcast}
+plot(nowcast, latest_obs = latest_germany_hosp)
+```
+
+## Plotting posterior predictions
+
+Plotting posterior predictions can be a useful way of assessing performance and checking that the model is capturing the underlying data generation process adequately. We can do this directly on the output of `epinowcast()` using
+
+```{r pp, fig.width = 16, fig.height = 16, message = FALSE}
+plot(nowcast, type = "posterior") +
+  facet_wrap(vars(reference_date), scale = "free")
+```
+
+## Using package functions rather than S3 methods
+
+Rather than using S3 methods supplied for `epinowcast()` directly, package functions can also be used to extract nowcast posterior samples, summarise them, and then plot them. This is demonstrated here by plotting the 7 day incidence for hospitalisations.
+
+```{r week_nowcast}
+# extract samples
+samples <- summary(nowcast, type = "nowcast_samples")
```

Comment:
This is S3 though.

Reply:
agreed - i'm not familiar enough with this part of interface / package conception to know how to fix it. this should definitely go in a cleaning pass of this vignette.

Reply:
I think it's a wording issue. Maybe the text is referring to using provided functions post `summary()`. Will note it down for upgrade.

Reply:
See linked issue.